src/store.js

```diff
@@ -0,0 +1,135 @@
+import { owl } from "@odoo/owl";
+import "./app.scss";
+
+const { Store } = owl;
+import {
+  originalBoard,
+  boardBottom,
+  boardLeftBorder,
+  boardRight,
+  boardTopBorder,
+} from "./constant";
+
+export function createTaskStore() {
+  const state = {
+    board: _.cloneDeep(originalBoard),
+    isActiveNote: { value: false },
+    aroundCell: [], // lưu vị trí các cell dọc ngang quanh cell được chọn
+    selectingX: null,
+    selectingY: null,
+    moves: [], // lưu các nước đi để hoàn tác
+    history: [],
+
+    boardHightlightAround: function () {
+      const matrices = [];
+      // Loop through rows
+      for (let i = 0; i < 9; i += 3) {
+        // Loop through columns
+        for (let j = 0; j < 9; j += 3) {
+          const matrix = [];
+          // Loop to create 3x3 matrix
+          for (let row = i; row < i + 3; row++) {
+            for (let col = j; col < j + 3; col++) {
+              matrix.push({ x: row, y: col });
+            }
+          }
+          matrices.push(matrix);
+        }
+      }
+
+      return matrices;
+    },
+  };
+  const actions = {
+    setBoard({ state }, position, fillNumber) {
+      state.board[position.x][position.y] = fillNumber;
+    },
+    undoBoard({state}, board) {
+      state.board = board ? _.cloneDeep(board) : _.cloneDeep(originalBoard)
+    },    
+    activeNoteMode({ state }) {
+      state.isActiveNote.value = !state.isActiveNote.value;
+    },
+    createAroundCell({ state }, position) {
+      const array = [];
+      if (position.x && position.y) {
+        for (let y = 0; y < 9; y++) {
+          if (y !== position.y) {
+            array.push({ x: position.x, y: y });
+          }
+        }
+
+        for (let x = 0; x < 9; x++) {
+          if (x !== position.x) {
+            array.push({ x: x, y: position.y });
+          }
+        }
+      }
+      state.aroundCell = array;
+    },
```

Comment:
chuyển mấy hàm check này sang getter giúp e. Và chuyển tiền tố check thành "is" giúp e. Cơ bản mấy hàm này không phải là actions, a cần phân biết sự khác nhau giữa action và getters

Reply:
đơn giản là như thế này, cái nào mà trả về giá trị mà giá trị đó chỉ phụ thuộc vào state thì là getter, còn khi nào a cần thực hiện 1 action để thay đổi state thì là action